packages/koenig-lexical/src/hooks/useVisibilitySettingsToggle.js

```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import {$getNodeByKey} from 'lexical';
+import {$isHtmlNode} from '../nodes/HtmlNode';
+import {DESELECT_CARD_COMMAND, EDIT_CARD_COMMAND, SELECT_CARD_COMMAND} from '../plugins/KoenigBehaviourPlugin';
+
+export function useVisibilitySettingsToggle(editor, nodeKey, isSelected, showVisibilitySettings, setShowVisibilitySettings, isEditing) {
+    return React.useCallback((event) => {
+        event.preventDefault();
+        event.stopPropagation();
+
+        editor.update(() => {
+            const cardNode = $getNodeByKey(nodeKey);
+
+            // If the card is an html card, we toggle the visibility settings differently
+            // because we want to show the visibility settings panel while in selected mode
+            // instead of entering edit mode
+            if ($isHtmlNode(cardNode)) {
+                setShowVisibilitySettings(showVisibilitySettings ? false : true);
+                if (!isSelected) {
+                    editor.dispatchCommand(SELECT_CARD_COMMAND, {cardKey: nodeKey, focusEditor: true});
+                }
```

Comment:
This feels like a code smell that's indicative of a pattern not being quite right. A "generic" hook should not typically have any specific knowledge of what is calling it or what it's acting upon.

If we stick with a hook approach then it would be better if `cardNode` exposed something that allows the hook to enter this conditional path, similar to the code below where it switches between edit/select based on `cardNode` exposing whether it has an edit mode.